### ISSUE TITLE
Incorrect use of depth texture attachments

### ISSUE THREAD

**Issue by vanruesc** [vanruesc/postprocessing#225]
### Description of the bug

The current depth texture system assumes that a depth texture can be assigned to a render target at any time. This assumption is incorrect as render targets are initialized once when they are used. Certain pass chains will cause the internal frame buffer to be initialized with a basic depth buffer attachment and the subsequent assignment of a depth texture won't have any effect. As a result, the depth texture won't contain the depth information of the rendered scene.

### To Reproduce

1. Add a `ClearPass` to the composer.
2. Add a `RenderPass` to the composer.
3. Add an `EffectPass` containing a depth-based effect such as `SSAOEffect` to the composer.
4. See the effect fail due to missing depth information.

### Expected behavior

Depth-based effects shouldn't fail with certain pass chains. One potential solution would be to assign the depth texture to a dedicated render target for geometry passes. Going one step further, a more complex render target management system could be implemented to allow passes to explicitly define input and output buffers.

### Library versions used

 - Three: [e.g. 0.120.0]
 - Post Processing: [e.g. 6.17.1]

**Comment by vanruesc** [comment]
Added a temporary fix in `postprocessing@6.17.2`.

**Comment by alexpineda** [comment]
I'm having this issue with 6.19.3. I'm not sure if it's something I'm doing something wrong on my end but seeing this ticket it seems to confirm my suspicion of missing depth information, even though the needsDepthTexture and the following logic execute correctly. My example follows closely to yours, RenderPass with clearPass enabled -> EffectPass with DepthEffect.

**Comment by vanruesc** [comment]
Depth texture management is fixed in the postprocessing v7 alpha which will eventually replace v6.

I'm keeping this ticket open until v7 lands in case someone stumbles over this bug.

**Comment by hybridherbst** [comment]
I believe we're also running into this issue. I can see there's ongoing changes to v7 – do you feel the beta is ready for testing?

**Comment by vanruesc** [comment]
Yes, v7 is ready for testing and has become much more stable since the first beta release.

There are currently three main issues that prevent a full release:
1. Removing [this onBeforeCompile crutch](https://github.com/pmndrs/postprocessing/blob/d32751a9457c2fa0b8fce6b1211234f5dd37f986/src/passes/GeometryPass.ts#L336-L389) (https://github.com/mrdoob/three.js/pull/27808).
2. Better management of [backgrounds/skyboxes](https://github.com/pmndrs/postprocessing/blob/d32751a9457c2fa0b8fce6b1211234f5dd37f986/src/utils/Background.ts) (for G-Buffer purposes).
3. Adding more essential Effects like DoF, SSAO, SSR.

**Comment by vanruesc** [comment]
Should be fixed in [v6.39.0](https://github.com/pmndrs/postprocessing/releases/tag/v6.39.0).
